app/jobs/waveform_job.rb

```diff
@@ -51,6 +48,22 @@ def file_uri(master_file)
       end
     end
 
+    def derivative_file_uri(master_file)
+      derivatives = master_file.derivatives
+      uri = nil
+
+      # Find the lowest quality stream
+      ['high', 'medium', 'low'].each do |quality|
+        d = derivatives.select { |d| d.quality == quality }.first
```

Comment:
You should be able to simplify this with `find`:
```suggestion
        d = derivatives.find { |derivative| derivative.quality == quality }
```

Reply:
Thanks!

Reply:
Nvm, that didn't work, I remember starting out with this now. 
```
Failure/Error: d = derivatives.find { |derivative| derivative.quality == quality }
     ArgumentError:
       Couldn't find Derivative without an ID
```

Reply:
Ah, probably needs to be `derivatives.to_a.find` due to AF.